### Zadanie 1.

Otrzymaliśmy dane w następującej postaci: "{dzień tygodnia};{cztery znaki zawierające wskazanie czujnika}" (zawsze są to 4 znaki!). Napisz skrypt, który wygeneruje raport z konkretnego dnia na prośbę użytkownika:

Poproś o podanie dnia dla którego ma zostać wyliczony raport (wielkość liter nie powinna mieć znaczenia)

Pobierz odpowiednią wartość z danych

Przelicz wskazanie czujnika na temperaturę dzieląc przez wartość z tabeli

Wypisz temperaturę z dokładnością do trzeciego miejsca po przecinku (tak jak na przykładzie) i ze znakiem specjalnym ℃ (\u2103)
    
BONUS. Do punktu 1 dodaj walidację - upewnij się, że użytkownik podał jeden z dni który możemy znaleźć w danych

BONUS. Do punktu 4 dodaj wizualizację temperatury w zakresie 0 do 100 stopni na pasku złożonym z 20 znaków (np. dla 50 stopni, zapełnione jest 10/20 znaków)


Wskazanie czujnika - przelicznik:

Większe niż, lub równe 1400 - 22.5

Pomiędzy 1400, a 1200 - 23.1

Mniejsze niż, lub równe 1200 - 23.5

In [26]:
data = """
monday;1250
tuesday;1405
wednesday;1750
thursday;1100
friday;0800
saturday;1225
sunday;1355
"""

In [27]:
#funkcja zapisująca daty do slownika z podanego zbioru
def data_to_dict(data):

    data_lines = data.strip().split('\n')
    data_dict = {}

    for line in data_lines:
        dat = line.split(';')
        data_dict[dat[0]] = int(dat[1])

    return data_dict

In [28]:
#funkcja pobierajace wartosc dla czytnika
def get_sensor_value(day):
    data_dict = data_to_dict(data)
    
    return data_dict[day]

In [29]:
#funkcja przeliczajaca temperature zgodnie z podanymi danymi
def calculate_temp(value):
    if value >= 1400:
        return value/22.5
    elif 1200 < value < 1400:
        return value/23.1
    else:
        return value/23.5

In [30]:
#funkcja do narysowania temperatury graficznie
def draw_graph(temperature):
    number_of_hash = int(temperature/100 * 20)
    graph = ''
    for index in range(22):
        if index == 0 or index == 21:
            graph += '|'
        elif index <= number_of_hash:
            graph += '#'
        else:
            graph += '-'
    return graph

In [31]:
#generowanie raportu
def generate_raport():
    input_day = input("Proszę o wprowadzenie poprawnego dnia tygodnia: ")
    day_to_check = str(input_day).lower()
    try:
        sensor_value = get_sensor_value(day_to_check)
        temperature_value  = calculate_temp(sensor_value)
        message = f'Dane dla "{day_to_check}"\nWskazanie czujnika: {sensor_value}\nTemperatura: {temperature_value:.03f}\u2103\n0\u2103 '
        message += draw_graph(temperature_value) # wizualizacja grafu - bonus nr 2
        message += ' 100\u2103'
        print (message)
    except:
        print("Proszę o wprowadzenie poprawnego dnia tygodnia: ") # sprawdzenie czy dzień jest poprawny

generate_raport()

Dane dla "monday"
Wskazanie czujnika: 1250
Temperatura: 54.113℃
0℃ |##########----------| 100℃


Przykładowe wyjście

Dane dla "thursday"

Wskazanie czujnika: 1100

Temperatura: 46.809℃

0℃ |#########-----------| 100℃  # BONUS

### Zadanie 2.

Pobierz zbiór danych o zawodnikach NBA: https://www.kaggle.com/justinas/nba-players-data
otwórz go za pomocą Pandas i odpowiedz na pytania:

a) Czy jest jakikolwiek zawodnik spoza USA, który przed NBA uczęszczał do college'u Kentucky?

b) Ile wynosi średni wzrost zawodnika w calach? Czy od pierwszego sezonu w tym zbiorze do teraz powiększył się czy zmniejszył?

c) Kto w badanym okresie rzucił łącznie najwięcej punktów? Czy ta osoba miała też największą średnią punktów na sezon?

d) W której rundzie draftu największy udział procentowy mieli zawodnicy ważący więcej niż 100 kg?

Wstepna analiza:

In [33]:
import pandas as pd 

In [34]:
df = pd.read_csv('nba/all_seasons.csv')

In [35]:
df.head()

Unnamed: 0        player_name team_abbreviation   age  player_height  \
0           0      Dennis Rodman               CHI  36.0         198.12   
1           1  Dwayne Schintzius               LAC  28.0         215.90   
2           2       Earl Cureton               TOR  39.0         205.74   
3           3        Ed O'Bannon               DAL  24.0         203.20   
4           4        Ed Pinckney               MIA  34.0         205.74   

   player_weight                      college country draft_year draft_round  \
0      99.790240  Southeastern Oklahoma State     USA       1986           2   
1     117.933920                      Florida     USA       1990           1   
2      95.254320                Detroit Mercy     USA       1979           3   
3     100.697424                         UCLA     USA       1995           1   
4     108.862080                    Villanova     USA       1985           1   

   ...  pts   reb  ast  net_rating  oreb_pct  dreb_pct  usg_pct  ts_pct  \
0  ...  5.7  16.1  3.1        16.1     0.186     0.323    0.100   0.479   
1  ...  2.3   1.5  0.3        12.3     0.078     0.151    0.175   0.430   
2  ...  0.8   1.0  0.4        -2.1     0.105     0.102    0.103   0.376   
3  ...  3.7   2.3  0.6        -8.7     0.060     0.149    0.167   0.399   
4  ...  2.4   2.4  0.2       -11.2     0.109     0.179    0.127   0.611   

   ast_pct   season  
0    0.113  1996-97  
1    0.048  1996-97  
2    0.148  1996-97  
3    0.077  1996-97  
4    0.040  1996-97  

[5 rows x 22 columns]

In [36]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12305 entries, 0 to 12304
Data columns (total 22 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         12305 non-null  int64  
 1   player_name        12305 non-null  object 
 2   team_abbreviation  12305 non-null  object 
 3   age                12305 non-null  float64
 4   player_height      12305 non-null  float64
 5   player_weight      12305 non-null  float64
 6   college            10516 non-null  object 
 7   country            12305 non-null  object 
 8   draft_year         12305 non-null  object 
 9   draft_round        12305 non-null  object 
 10  draft_number       12305 non-null  object 
 11  gp                 12305 non-null  int64  
 12  pts                12305 non-null  float64
 13  reb                12305 non-null  float64
 14  ast                12305 non-null  float64
 15  net_rating         12305 non-null  float64
 16  oreb_pct           123

In [37]:
df.describe()

Unnamed: 0           age  player_height  player_weight            gp  \
count  12305.000000  12305.000000   12305.000000   12305.000000  12305.000000   
mean    6152.000000     27.084518     200.611602     100.369926     51.290532   
std     3552.291866      4.335868       9.146321      12.477150     25.095909   
min        0.000000     18.000000     160.020000      60.327736      1.000000   
25%     3076.000000     24.000000     193.040000      90.718400     31.000000   
50%     6152.000000     26.000000     200.660000      99.790240     57.000000   
75%     9228.000000     30.000000     208.280000     108.862080     73.000000   
max    12304.000000     44.000000     231.140000     163.293120     85.000000   

                pts           reb           ast    net_rating      oreb_pct  \
count  12305.000000  12305.000000  12305.000000  12305.000000  12305.000000   
mean       8.172775      3.559155      1.813986     -2.255733      0.054473   
std        5.974957      2.483550      1.794155     12.673254      0.043599   
min        0.000000      0.000000      0.000000   -250.000000      0.000000   
25%        3.600000      1.800000      0.600000     -6.400000      0.021000   
50%        6.700000      3.000000      1.200000     -1.300000      0.041000   
75%       11.500000      4.700000      2.400000      3.200000      0.084000   
max       36.100000     16.300000     11.700000    300.000000      1.000000   

           dreb_pct       usg_pct        ts_pct       ast_pct  
count  12305.000000  12305.000000  12305.000000  12305.000000  
mean       0.141014      0.184891      0.511060      0.131358  
std        0.062704      0.053390      0.101163      0.094483  
min        0.000000      0.000000      0.000000      0.000000  
25%        0.096000      0.149000      0.480000      0.066000  
50%        0.131000      0.181000      0.524000      0.103000  
75%        0.180000      0.217000      0.561000      0.178000  
max        1.000000      1.000000      1.500000      1.000000

In [38]:
df.columns

Index(['Unnamed: 0', 'player_name', 'team_abbreviation', 'age',
       'player_height', 'player_weight', 'college', 'country', 'draft_year',
       'draft_round', 'draft_number', 'gp', 'pts', 'reb', 'ast', 'net_rating',
       'oreb_pct', 'dreb_pct', 'usg_pct', 'ts_pct', 'ast_pct', 'season'],
      dtype='object')

In [39]:
cols = ['player_name', 'team_abbreviation', 'age',
       'player_height', 'player_weight', 'college', 'country', 'draft_year',
       'draft_round', 'draft_number', 'gp', 'pts', 'reb', 'ast', 'net_rating',
       'oreb_pct', 'dreb_pct', 'usg_pct', 'ts_pct', 'ast_pct', 'season']

In [40]:
for i in cols:
    print(f'Liczba kategorii zmiennej {i}: \n {df[i].value_counts()}')

Liczba kategorii zmiennej player_name: 
 player_name
Vince Carter        22
Dirk Nowitzki       21
Kevin Garnett       20
Jamal Crawford      20
Kobe Bryant         20
                    ..
JaJuan Johnson       1
Carldell Johnson     1
Derrick Byars        1
Eric Dawson          1
Micah Potter         1
Name: count, Length: 2463, dtype: int64
Liczba kategorii zmiennej team_abbreviation: 
 team_abbreviation
CLE    433
TOR    428
MIA    426
LAC    424
DAL    422
WAS    422
ATL    421
PHI    420
HOU    418
SAS    413
DEN    412
ORL    411
LAL    411
NYK    410
IND    410
MIL    409
GSW    409
POR    407
BOS    407
CHI    406
DET    400
MIN    399
PHX    399
SAC    398
UTA    397
MEM    352
CHA    288
NJN    257
OKC    239
SEA    182
BKN    180
NOP    159
NOH    143
CHH     89
VAN     72
NOK     32
Name: count, dtype: int64
Liczba kategorii zmiennej age: 
 age
24.0    1277
23.0    1164
25.0    1141
26.0    1015
27.0     972
28.0     868
29.0     785
22.0     771
30.0     711
31.0     620


a) Czy jest jakikolwiek zawodnik spoza USA, który przed NBA uczęszczał do college'u Kentucky?

In [41]:
df[list((df.country != 'USA') & (df.college == 'Kentucky'))]

Unnamed: 0              player_name team_abbreviation   age  \
1910         1910          Jamaal Magloire               CHH  23.0   
2529         2529          Jamaal Magloire               CHH  24.0   
2799         2799          Jamaal Magloire               NOH  25.0   
3416         3416          Jamaal Magloire               NOH  26.0   
3803         3803          Jamaal Magloire               NOH  27.0   
4074         4074          Jamaal Magloire               MIL  28.0   
4438         4438          Jamaal Magloire               POR  29.0   
4485         4485         Kelenna Azubuike               GSW  23.0   
4940         4940          Jamaal Magloire               DAL  30.0   
5204         5204         Kelenna Azubuike               GSW  24.0   
5366         5366          Jamaal Magloire               MIA  31.0   
5573         5573         Kelenna Azubuike               GSW  25.0   
6087         6087         Kelenna Azubuike               GSW  26.0   
6162         6162          Jamaal Magloire               MIA  32.0   
6656         6656          Jamaal Magloire               MIA  33.0   
6696         6696          Jamaal Magloire               TOR  34.0   
6826         6826              Enes Kanter               UTA  20.0   
7089         7089         Kelenna Azubuike               DAL  28.0   
7398         7398              Enes Kanter               UTA  21.0   
8023         8023              Enes Kanter               UTA  22.0   
8222         8222              Enes Kanter               OKC  23.0   
8838         8838              Enes Kanter               OKC  24.0   
9187         9187          Skal Labissiere               SAC  21.0   
9500         9500             Jamal Murray               DEN  20.0   
9516         9516              Enes Kanter               OKC  25.0   
9711         9711             Jamal Murray               DEN  21.0   
9939         9939          Skal Labissiere               SAC  22.0   
10144       10144          Skal Labissiere               POR  23.0   
10150       10150  Shai Gilgeous-Alexander               LAC  20.0   
10313       10313          Isaac Humphries               ATL  21.0   
10570       10570             Jamal Murray               DEN  22.0   
10677       10677            Mychal Mulder               GSW  26.0   
10767       10767               Trey Lyles               SAS  24.0   
10813       10813           Wenyen Gabriel               POR  23.0   
10874       10874          Skal Labissiere               ATL  24.0   
10879       10879  Shai Gilgeous-Alexander               OKC  21.0   
11043       11043             Jamal Murray               DEN  23.0   
11210       11210            Mychal Mulder               GSW  27.0   
11353       11353           Wenyen Gabriel               NOP  24.0   
11371       11371               Trey Lyles               SAS  25.0   
11414       11414            Nick Richards               CHA  23.0   
11443       11443  Shai Gilgeous-Alexander               OKC  22.0   
11615       11615             Jamal Murray               DEN  24.0   
12033       12033  Shai Gilgeous-Alexander               OKC  23.0   
12120       12120           Wenyen Gabriel               LAL  25.0   
12133       12133               Trey Lyles               SAC  26.0   
12153       12153             Olivier Sarr               OKC  23.0   
12231       12231            Mychal Mulder               MIA  28.0   
12257       12257            Nick Richards               CHA  24.0   

       player_height  player_weight   college      country draft_year  \
1910          208.28     117.933920  Kentucky       Canada       2000   
2529          210.82     117.480328  Kentucky       Canada       2000   
2799          210.82     117.480328  Kentucky       Canada       2000   
3416          210.82     117.480328  Kentucky       Canada       2000   
3803          210.82     117.480328  Kentucky       Canada       2000   
4074          210.82     117.480328  Kentucky       Canada       200

In [42]:
df[list((df.country != 'USA') & (df.college == 'Kentucky'))].describe()

Unnamed: 0        age  player_height  player_weight         gp  \
count     49.000000  49.000000      49.000000      49.000000  49.000000   
mean    8488.775510  24.653061     204.547755     104.585355  50.734694   
std     3098.279242   3.351312       7.983544      12.496782  26.397424   
min     1910.000000  20.000000     190.500000      81.646560   3.000000   
25%     6087.000000  23.000000     195.580000      93.893544  26.000000   
50%     9516.000000  24.000000     208.280000     106.140528  56.000000   
75%    11043.000000  26.000000     210.820000     117.480328  75.000000   
max    12257.000000  34.000000     213.360000     121.109064  82.000000   

             pts        reb        ast  net_rating   oreb_pct   dreb_pct  \
count  49.000000  49.000000  49.000000   49.000000  49.000000  49.000000   
mean    9.044898   4.416327   1.293878   -2.581633   0.075061   0.169163   
std     6.145495   2.335536   1.561971    6.972526   0.046314   0.070703   
min     0.800000   0.000000   0.000000  -22.100000   0.000000   0.000000   
25%     4.100000   2.900000   0.400000   -7.800000   0.026000   0.114000   
50%     8.500000   4.000000   0.700000   -1.900000   0.078000   0.155000   
75%    12.700000   5.100000   1.200000    3.200000   0.111000   0.223000   
max    24.500000  10.300000   5.900000    8.000000   0.168000   0.352000   

         usg_pct     ts_pct    ast_pct  
count  49.000000  49.000000  49.000000  
mean    0.186571   0.545592   0.080633  
std     0.050233   0.068550   0.069629  
min     0.103000   0.357000   0.000000  
25%     0.147000   0.523000   0.043000  
50%     0.177000   0.554000   0.060000  
75%     0.220000   0.588000   0.078000  
max     0.305000   0.689000   0.311000

Odpowiedz: 49 graczy jest spoza USA i takich co uczęszczli na Uniwersytet Kentucky

b) Ile wynosi średni wzrost zawodnika w calach? Czy od pierwszego sezonu w tym zbiorze do teraz powiększył się czy zmniejszył?

In [43]:
(df['player_height'].mean()) / 2.54

78.9809455857839

Średni wzrost zawodnika w calach wynosi 78.98 "

In [44]:
df.groupby("season")[["player_height"]].mean()

player_height
season                
1996-97     200.994830
1997-98     200.619499
1998-99     200.903007
1999-00     200.920959
2000-01     200.752154
2001-02     201.017909
2002-03     201.520514
2003-04     201.315113
2004-05     201.245733
2005-06     200.920655
2006-07     200.704367
2007-08     200.851486
2008-09     201.196539
2009-10     200.873394
2010-11     201.217832
2011-12     200.832343
2012-13     200.769765
2013-14     200.772946
2014-15     200.838455
2015-16     200.949034
2016-17     200.916790
2017-18     200.457741
2018-19     200.535396
2019-20     198.941059
2020-21     198.914926
2021-22     198.350909

Pierwszy sezon w bazie: 200.995

Ostatni sezon w bazie: 198.350

Sredni wzrost zawodników spadł

c) Kto w badanym okresie rzucił łącznie najwięcej punktów? Czy ta osoba miała też największą średnią punktów na sezon?

In [45]:
df.groupby("player_name")["pts"].sum().sort_values(ascending=False)

player_name
LeBron James        515.1
Kobe Bryant         484.0
Dirk Nowitzki       422.3
Carmelo Anthony     418.3
Kevin Durant        379.4
                    ...  
Ibrahim Kutluay       0.0
Guy Rucker            0.0
Greg Whittington      0.0
Gani Lawal            0.0
Jon Teske             0.0
Name: pts, Length: 2463, dtype: float64

LeBron James rzucił najwięcej puntków; 515.1

In [46]:
df.groupby("player_name")["pts"].mean().sort_values(ascending=False)

player_name
LeBron James      27.110526
Kevin Durant      27.100000
Luka Doncic       26.525000
Allen Iverson     26.064286
Trae Young        25.600000
                    ...    
William Howard     0.000000
JJ O'Brien         0.000000
Cuonzo Martin      0.000000
Royce White        0.000000
Jon Teske          0.000000
Name: pts, Length: 2463, dtype: float64

Tak - największa śrenia LeBron James 27.11

d) W której rundzie draftu największy udział procentowy mieli zawodnicy ważący więcej niż 100 kg?

In [47]:
df_w = df[df.player_weight > 100]
df_w

Unnamed: 0        player_name team_abbreviation   age  player_height  \
1               1  Dwayne Schintzius               LAC  28.0         215.90   
3               3        Ed O'Bannon               DAL  24.0         203.20   
4               4        Ed Pinckney               MIA  34.0         205.74   
7               7     Elden Campbell               LAL  28.0         213.36   
11             11     Elmore Spencer               SEA  27.0         213.36   
...           ...                ...               ...   ...            ...   
12294       12294      Mason Plumlee               CHA  32.0         210.82   
12296       12296  Marvin Bagley III               DET  23.0         210.82   
12297       12297    Marquese Chriss               DAL  24.0         205.74   
12300       12300    Markieff Morris               MIA  32.0         205.74   
12304       12304       Micah Potter               DET  24.0         208.28   

       player_weight           college country draft_year draft_round  ...  \
1         117.933920           Florida     USA       1990           1  ...   
3         100.697424              UCLA     USA       1995           1  ...   
4         108.862080         Villanova     USA       1985           1  ...   
7         113.398000           Clemson     USA       1990           1  ...   
11        122.469840  Nevada-Las Vegas     USA       1992           1  ...   
...              ...               ...     ...        ...         ...  ...   
12294     115.212368              Duke     USA       2013           1  ...   
12296     106.594120              Duke     USA       2018           1  ...   
12297     108.862080        Washington     USA       2016           1  ...   
12300     111.130040            Kansas     USA       2011           1  ...   
12304     112.490816         Wisconsin     USA  Undrafted   Undrafted  ...   

        pts  reb  ast  net_rating  oreb_pct  dreb_pct  usg_pct  ts_pct  \
1       2.3  1.5  0.3        12.3     0.078     0.151    0.175   0.430   
3       3.7  2.3  0.6        -8.7     0.060     0.149    0.167   0.399   
4       2.4  2.4  0.2       -11.2     0.109     0.179    0.127   0.611   
7      14.9  8.0  1.6         3.3     0.095     0.183    0.222   0.520   
11      0.0  0.0  0.0      -147.5     0.000     0.000    0.126   0.000   
...     ...  ...  ...         ...       ...       ...      ...     ...   
12294   6.5  7.7  3.1         0.9     0.091     0.205    0.117   0.603   
12296  11.3  7.0  0.8        -5.6     0.084     0.200    0.191   0.551   
12297   4.5  3.0  0.5        -2.1     0.096     0.197    0.205   0.518   
12300   7.6  2.6  1.4         4.5     0.059     0.089    0.197   0.547   
12304   4.0  3.0  0.0       -56.4     0.095     0.125    0.148   0.505   

       ast_pct   season  
1        0.048  1996-97  
3        0.077  1996-97  
4        0.040  1996-97  
7        0.087  1996-97  
11       0.000  1996-97  
...        ...      ...  
12294    0.165  2021-22  
12296    0.049  2021-22  
12297    0.079  2021-22  
12300    0.116  2021-22  
12304    0.000  2021-22  

[5938 rows x 22 columns]

In [48]:
(df_w["draft_round"].value_counts() / df["draft_round"].value_counts()).sort_values(ascending=False)

draft_round
4            0.750000
1            0.524113
2            0.486942
3            0.400000
6            0.400000
7            0.400000
Undrafted    0.349164
0                 NaN
8                 NaN
Name: count, dtype: float64

Największy udział procentowy mieli zawodnicy ważący więcej niż 100 kg w drafcie nr 4 - 75 %

### Zadanie 3.

Napisz funkcję create_tower, która jako input przyjmuje integera większego od 0
i zwraca listę, która tworzy "piramidę" z tyloma piętrami:
Przykładowe rozwiązanie:
In: create_tower(3)
Out:
[
  '  *  ',
  ' *** ',
  '*****'
]

In [49]:
floors = input("Podaj liczbę całkowitą pięter, większą od 0")

def create_tower(floors):
    try: 
        if int(floors) > 0:
            fl = int(floors)
            for i in range(1, fl + 1):
                print((fl-i)*" "+ i*"*" + (i-1)*"*" + (fl-i)*" ")
                i+=1
        else:
            print("Proszę podać liczbę całkowitą większą od zera") 
    except:
        print("Proszę podać liczbę całkowitą większą od zera")

create_tower(floors)

     *     
    ***    
   *****   
  *******  
 ********* 
***********


### Zadanie 4
. Bonusowe

Napisz gre "kółko i krzyżyk"

Wyświetlaj aktualny stan planszy przy pomocy znaków 'o', 'x', '-' w trzech wierszach Np.:

--x

-x-

oo-

Naprzemiennie pytaj gdzie postawić 'x', a gdzie 'o' przyjmując numer pozycji od 1 do 9 Tzn.:

123

456

789

Zakończ grę gdy jedna ze stron wygra, lub wszystkie miejsca będą zapełnione

Na razie wyświetlaj wszystko po kolei, jedno pod drugim

In [50]:
#nadanie tablicy
def initialize_board():
    return {1: ["-","-","-"],
            2: ["-","-","-"],
            3: ["-","-","-"]}

#funkcja wizualizujaca tablice
def visualize_board(board):
    for pos in board:
        row = ''
        for value in board[pos]:
            row += value
        print(row)
    print("")

#funkcja ktora opisuje pozycje - z pozycji okresla kroty rzad i kolumna w naszej tablicy odpowiada lokalizacji
def get_row_column(position):
    row = (position - 1) // 3 + 1
    if row == 1:
        column = position - row
    elif row == 2:
        column = position - 4
    else: 
        column = position - 7
    return [row, column]

#funkcja ktora nadpisuje nam "-" w naszej tablicy na x i o
def insert_symbol_to_position(board, symbol, position):
    indexes = get_row_column(position)
    symbol_to_update = board[indexes[0]][indexes[1]]
    if symbol_to_update == '-':
        board[indexes[0]][indexes[1]] = symbol
        return True
    return False

#funckja definiująca kolejnosc symboli x i o
def get_symbol(round):
    symbol = ''
    if round % 2 == 0:
        symbol += 'x'
    elif round % 2 != 0:
        symbol += 'o'
    return symbol

#funckja weryfikujaca wygrana
def verify_if_won(position, board):
    indexes = get_row_column(position)
    row = indexes[0]
    column = indexes[1]
    if((board[row][0] == board[row][1]) & (board[row][0] == board[row][2])):
        return True
    if((board[1][column] == board[2][column]) & (board[1][column] == board[3][column])):
        return True
    if board[2][1] == '-':
        return False
    if((board[1][0] == board[2][1]) & (board[1][0] == board[3][2])):
        return True
    if((board[1][2] == board[2][1]) & (board[1][2] == board[3][0])):
        return True

def tic_tac_toe():
    board = initialize_board()

    round = 0
    while round < 9:
        try: 
            symbol = get_symbol(round)

            input_position = int(input(f'Where do you want to put "{symbol}": '))

            if input_position < 1 or input_position > 9:
                print('Wrong position')
                continue

            is_inserted = insert_symbol_to_position(board, symbol, int(input_position))
                
            if is_inserted == False:
                print('This place is already taken, try again')
                continue

            won = verify_if_won(int(input_position),board)
            if won:
                visualize_board(board=board)
                print(f'{symbol} won the round')
                break

            round += 1
        except:
            print("Wrong input")

        visualize_board(board=board)

tic_tac_toe()

--x
---
---

-ox
---
---

-ox
x--
---

oox
x--
---

oox
xx-
---

oox
xxo
---

oox
xxo
x--

x won the round
